TheBeyondGroup/shopkeeper
# Set up the project to run through the CLI in development

This PR adjusts the project set up for the Shopify CLI (a `devDependency`) to load the plugin automatically. Here are the changes that were necessary:

- A `postinstall` script at `bin/patch-cli.js` that includes `shopkeeper` as a plugin in the CLI's `package.json`. The need for this will eventually go away.
- I updated the project to use ESM instead of CJS as the module system. This aligns with the module system used by the CLI, CLI-Kit, and the rest of the plugins.
- Removed empty `args = {}` from the commands because it causes compilation issues with Typescript.

### How to test the changes

1. Pull the changes from this branch.
2. Delete `node_modules` and run `npm install`
3. Build the project with `npm run build`
4. Run `npm exec -- shopify --help`. The plugin commands should show up.
